src/main/java/com/shapesecurity/salvation/data/Policy.java

```diff
@@ -510,4 +510,24 @@ public boolean allowsStyleWithNonce(@Nonnull String nonce) {
     public boolean allowsStyleWithNonce(@Nonnull Base64Value nonce) {
         return this.allowsStyleWithNonce(nonce.value);
     }
+
+    public boolean allowsChildFromSource(@Nonnull URI source) {
+        ChildSrcDirective childSrcDirective = this.getDirectiveByType(ChildSrcDirective.class);
+        if (childSrcDirective == null) {
+            return this.defaultsAllowSource(source);
+        }
+        return childSrcDirective.matchesSource(this.origin, source);
+    }
+
+    public boolean allowsFrameFromSource(@Nonnull URI source) {
+        FrameSrcDirective frameSrcDirective = this.getDirectiveByType(FrameSrcDirective.class);
+        if (frameSrcDirective == null) {
+            // http://www.w3.org/TR/CSP2/#directive-frame-src
```

Comment:
I don't think this comment is necessary. None of the other directives that fall back to another set of sources (default sources) quote the spec.